crates/storage/provider/src/block.rs

```diff
@@ -168,15 +166,25 @@ pub fn insert_canonical_block<'a, TX: DbTxMut<'a> + DbTx<'a>>(
         let rec_tx = transaction.clone().into_ecrecovered().unwrap();
         tx.put::<tables::TxSenders>(current_tx_id, rec_tx.signer())?;
         tx.put::<tables::Transactions>(current_tx_id, rec_tx.into())?;
+        transition_id += 1;
```

Comment:
Was this a bug? We had to increment it before writing?

Reply:
Order was changed, before it was incremented at lines above.